src/CSharpLanguageServer/Server.fs

```diff
@@ -686,6 +686,43 @@ let setupServerHandlers settings (lspClient: LspClient) =
                 None |> success |> async.Return
     }
 
+    let handleTextDocumentTypeDefinition (scope: ServerRequestScope) (def: Types.TextDocumentPositionParams) : AsyncLspResult<Types.GotoResult option> = async {
+        let docMaybe = scope.GetAnyDocumentForUri def.TextDocument.Uri
+
+        return!
+            match docMaybe with
+            | Some doc -> async {
+                let! ct = Async.CancellationToken
+                let! sourceText = doc.GetTextAsync(ct) |> Async.AwaitTask
+                let position = sourceText.Lines.GetPosition(LinePosition(def.Position.Line, def.Position.Character))
+                let! symbolMaybe = SymbolFinder.FindSymbolAtPositionAsync(doc, position, ct) |> Async.AwaitTask
+
+                let symbols =
+                    match Option.ofObj symbolMaybe with
+                    | Some sym -> [sym]
+                    | None -> []
+
+                let typeSymbols = 
+                    match symbols with 
+                    | [] -> []
+                    | _  ->
+                        match symbols.Head with
+                        | :? ILocalSymbol as loc -> [loc.Type]
+                        | :? IFieldSymbol as field -> [field.Type]
+                        | :? IPropertySymbol as prop -> [prop.Type]
+                        | :? IParameterSymbol as param -> [param.Type]
+                        | _ -> []
+
+                let! locations =
+                     scope.ResolveTypeSymbolLocations doc.Project typeSymbols 
+
+                return locations |> Array.ofSeq |> GotoResult.Multiple |> Some |> success
```

Comment:
I don't think it's possible to have multiple locations for a type definition, but this made it easier to reuse existing definition code

Reply:
lgtm!